### PR TITLE
dac_dds test fixes

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

This is a collection of small fixes to the dac_dds testl

### Testing procedure

* Get a board with audio output (mcb2388 is the only currently supported; I'm running it on a WIP port on thingy52, and the STM32F4Discovery board could probably be enabled easily)
* Run tests/driver_dac_dds on it
* Execute things like `sine 1 1` (doesn't crash any more), `init` (doesn't print "argc: 1" but just the usage message), tap the microphone output (it's in a DC-free state from startup and not only after having run the greeting)

### Issues/PRs references

I reviewed and merged #13908, so I guess I get to fix what I didn't complain about in time ... but at least I get to ask @benpicco to review it.

### CODE REVIEW

File: tests/driver_dac_dds/main.c [hunk 1]
```diff
@@ -312,7 +329,9 @@ static const shell_command_t shell_commands[] = {
 
 int main(void)
 {
-    dac_init(DAC_DDS_PARAM_DAC);
+    dac_init(DAC_DDS_DAC);
+    /* Initialize to the idle level of 16bit audio */
+    dac_set(DAC_DDS_DAC, 1 << 15);
```
Comment (benpicco) [inline tests/driver_dac_dds/main.c:334]: maybe `dac_dds_init()` should take care of that?

**Review by benpicco — COMMENTED** [review]
Thank you for the cleanup!

File: tests/driver_dac_dds/main.c (hunk 1 above)
Comment (chrysn) [reply tests/driver_dac_dds/main.c:334]: Possibly -- but the question would come back with dac_init and dac_poweron/poweroff, and eventually to the question of turning on and off the power amplifier. Leaving dac_dds to be a tool that feeds a DAC with whichever values are passed in seems to be a clean cut of responsibilities -- every part that implies that the DAC data is audio would be left in the application.

File: tests/driver_dac_dds/main.c [hunk 2]
```diff
@@ -171,9 +178,17 @@ static void play_function(uint16_t period, uint32_t samples, sample_gen_t fun)
...
+    if (period > DAC_BUF_SIZE) {
+        printf("Period duration exceeds sample buffer size.\n");
+        return;
+    }
+
     /* only work with whole wave periods */
     uint16_t len_aligned = DAC_BUF_SIZE - DAC_BUF_SIZE % period;
 
+    /* One underrun indication is expected (for the first sample) */
+    int32_t underruns = -1;
```
Comment (benpicco) [inline tests/driver_dac_dds/main.c:190]: might as well use an `int` here, then you don't have to use those ugly `PRId32` format strings static tests wants you to use.

Comment (chrysn) [reply tests/driver_dac_dds/main.c:190]: Good point. An earlier version printed the samples too, but now that it doesn't, why bother.

Comment (chrysn) [reply tests/driver_dac_dds/main.c]: ... and fixup'd.

Comment (benpicco) [reply tests/driver_dac_dds/main.c]: please squash!

**Review by benpicco — APPROVED** [review]

**Pull request merged**